src/mono/wasm/runtime/crypto-worker.ts

```diff
@@ -299,33 +294,40 @@ class LibraryChannel {
     private read_response(): string {
         let response = "";
         for (; ;) {
-            const state = this.wait_for_state(state => state == this.STATE_RESP || state == this.STATE_RESP_P, "read_response");
-            this.acquire_lock();
-
-            try {
+            this.wait_for_state_change_to(state => state == this.STATE_RESP || state == this.STATE_RESP_P, "read_response");
+            let state;
```

Comment:
Does this variable need to be outside of the `using_lock` callback? It doesn't appear to be used outside of it.